src/hooks/useOutsideAlerter.js

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+
+export default (ref, onClickOutside) => {
+  function handleClickOutside(event) {
+    if (ref.current && !ref.current.contains(event.target)) {
+      onClickOutside();
+    }
+  }
+
+  useEffect(() => {
```

Comment:
If you don't declare a dependencies array on the effect this will get re-triggered on every render infinitely.
Do you have the hooks rules setup for eslint?

Reply:
I have the rules running yes, but it doesn't complain about this hook